container.go

```diff
@@ -50,6 +53,20 @@ type Container struct {
 	graph.Graph
 }
 
+// InvokeOnce only allows function invocation once to register the
+// return types. If return types are already registered, specific error
+// is returned to the caller
+func (c *Container) InvokeOnce(t interface{}) error {
+	ctype := reflect.TypeOf(t)
+	if ctype.Kind() != reflect.Func {
+		return errors.Wrapf(errParamType, _invokeErr, ctype)
+	}
+	if err := c.ValidateInvokeReturnTypes(ctype); err != nil {
+		return ErrInvokeOnce
```

Comment:
True. It does simplify the check and decision making for caller. Although the caller has access to the function signature to log it, or error out.